core/drivers/clk/clk_dt.c

```diff
@@ -134,20 +137,21 @@ static void parse_assigned_clock(const void *fdt, int nodeoffset)
 	unsigned long rate = 0;
 	struct clk *parent = NULL;
 	const uint32_t *rate_prop = NULL;
+	TEE_Result res = TEE_ERROR_GENERIC;
 
 	rate_prop = fdt_getprop(fdt, nodeoffset, "assigned-clock-rates",
 				&rate_len);
 	rate_len /= sizeof(uint32_t);
 
 	while (1) {
 		clk = clk_dt_get_by_idx_prop("assigned-clocks", fdt, nodeoffset,
-					     clock_idx);
-		if (!clk)
+					     clock_idx, &res);
+		if (res || !clk)
 			return;
 
 		parent = clk_dt_get_by_idx_prop("assigned-clock-parents", fdt,
-						nodeoffset, clock_idx);
-		if (parent) {
+						nodeoffset, clock_idx, &res);
+		if (!res && parent) {
```

Comment:
Same here, checking `parent` should be enough.

Reply:
ok